<?xml version="1.0"?>
<launch> 

  <group ns="spinning_lidar">
    <node name="nodelet_manager" pkg="nodelet" type="nodelet" args="manager" output="screen"/>

    <node name="laser_range_filter" pkg="nodelet" type="nodelet" args="load spinning_lidar_utils/LaserRangeFilterNodelet nodelet_manager" output="screen">
      <rosparam command="load"            file="$(find spinning_lidar_utils)/config/laser_range_filter.yaml"/>
    </node>

    <node name="ros_laser_assembler" pkg="laser_assembler" type="laser_scan_assembler">
      <rosparam command="load"            file="$(find spinning_lidar_utils)/config/ros_laser_assembler.yaml"/>
      <remap    from="scan"               to="filtered_scan"/>
    </node>

    <node name="interrupt_laser_assembler" pkg="nodelet" type="nodelet" args="load spinning_lidar_utils/InterruptLaserAssemblerNodelet nodelet_manager" output="screen">
      <rosparam command="load"            file="$(find spinning_lidar_utils)/config/interrupt_laser_assembler.yaml"/>
    </node>

    <node name="scan_to_cloud_converter" pkg="nodelet" type="nodelet" args="load spinning_lidar_utils/ScanToCloudConverterNodelet nodelet_manager" output="screen">
      <rosparam command="load"            file="$(find spinning_lidar_utils)/config/scan_to_cloud_converter.yaml"/>
    </node>
  </group>

</launch>